micromanager_gui/_gui_objects/_group_preset_table_widget.py

```diff
@@ -0,0 +1,120 @@
+from typing import Any
+
+from magicgui.widgets import ComboBox, FloatSlider, LineEdit, Table, Widget
+from qtpy import QtWidgets as QtW
+from qtpy.QtWidgets import QVBoxLayout
+
+from .. import _core
+
+
+class MainTable(Table):
+    def __init__(self) -> None:
+        super().__init__()
+        hdr = self.native.horizontalHeader()
+        hdr.setSectionResizeMode(hdr.Stretch)
+        vh = self.native.verticalHeader()
+        vh.setVisible(False)
+        vh.setSectionResizeMode(vh.Fixed)
+        vh.setDefaultSectionSize(24)
+        self.native.setEditTriggers(QtW.QTableWidget.NoEditTriggers)
+
+
+class MMGroupPresetTableWidget(QtW.QWidget):
+    def __init__(self):
+        super().__init__()
+
+        self._mmc = _core.get_core_singleton()
+        self._mmc.events.systemConfigurationLoaded.connect(self._populate_table)
```

Comment:
Definitely agree that we should be updating based on those signals. One thing to consider is what to do when we get a configGroupChanged event with a blank string for a preset. That can happen if the system state is changed such that it doesn't correspond to any preset (e.g. contains exposure and exposure was changed).  micromanager takes the approach of just putting a blank value in the dropdown which I always found to be confusing. In fact I actually thought it was a bug for a long time.

What do oyu think about doing the following when receiving a `configGroupChanged` signal with a blank string for the preset:

Change the visual indicators on the preset combobox to indicate that it **used** to be something but that that no longer exactly represents the state of the scope. I'm thinking something along the lines of graying out the text a little, or a red background, or maybe a strikethrough and also adding a tooltip that explains the situation. 

I wonder if we can be a bit more expressive in the UI here than micromanager is. In micromanager you can set a config group, but then if you change a property so that the system state no longer reflects that preset then the preset dropdown goes blank which was always a source of confusion for me. An easy way to end up there is if you put the exposure value into a preset. I now know that you shouldn't, but I suspect I'm not alone in having done that. The consequence is that if then change the exposure the configGroupChagned` signal will fire with an empty string as the preset.

So there's this splitting of state between what the user set, and what the actual state of the system is. I think we can represent that by instead of just showing preset names.

Reply:
See also the discussion of the two config signals here: https://github.com/tlambert03/pymmcore-plus/pull/38#issue-962948450 I think this is the relevant use case of the configSet signal. You set the config, but due to trickiness of what the devices do inside themselves the actual changed signal that gets fired is blank. Unless we are also listening to the set signal then you can up with the bad user experience of selecting something from a dropdown and having it immediately switch to blank.